Platon/backend/app/follow_system/views.py

```diff
@@ -492,5 +505,118 @@ def delete(user_id, self):
         else:
             return make_response(jsonify({'error': 'Input Format Error'}), 400)
 
+@follow_system_ns.route("/report")
+class ReportAPI(Resource):
+    '''
+        Endpoints for Report Model.
+    '''
+
+    @api.doc(responses={401: 'Account Problems', 400: 'Input Format Error', 500: ' Database Connection Error',
+                        404: 'Not found'})
+    @api.response(200, 'Success', report_list_model)
+    @api.expect(get_report_parser)
+    @login_required
+    @follow_required(param_loc='args', requested_user_id_key='reported_user_id')
+    def get(user_id, self):
+        '''
+            Get Reports of User.
+        '''
+        form = GetReportsForm(request.args)
+        if form.validate():
+            try:
+                user_reports = Reports.query.filter(
+                    Reports.reported_user_id == form.reported_user_id.data).all()
+            except:
+                return make_response(jsonify({'error': 'Database Connection Error'}), 500)
+            if len(user_reports) == 0:
+                return make_response(jsonify({'result': []}), 200)
+            return_list = []
+            for report in user_reports:
+                try:
+                    reporter = User.query.filter(User.id == report.owner_id).first()
+                except:
+                    return make_response(jsonify({'error': 'Database Connection Error'}), 500)
+                if reporter is None:
+                    return make_response(jsonify({'error': 'Reporter is None, smth wrong'}), 500)
+                return_list.append({
+                    "report_id": report.id,
+                    "owner_id": report.owner_id,
```

Comment:
I think reporting should be anonymous. Returning owner information may prevent them to take action.

Reply:
You are right, i will remove reporter information from response.